src/D3MHub.sol

```diff
@@ -395,128 +395,47 @@ contract D3MHub {
         @param ilk bytes32 of the D3M ilk name
     */
     function exec(bytes32 ilk) external lock {
+        // IMPORTANT: this function assumes Vat rate of D3M ilks will always be == 1 * RAY (no fees).
+        // That's why this module converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
+
         (uint256 Art, uint256 rate, uint256 spot, uint256 line,) = vat.ilks(ilk);
         require(rate == RAY, "D3MHub/rate-not-one");
         require(spot == RAY, "D3MHub/spot-not-one");
 
         ID3MPool _pool = ilks[ilk].pool;
 
         _pool.preDebtChange("exec");
-        uint256 currentAssets = _pool.assetBalance();
 
         if (vat.live() == 0) {
             // MCD caged
-            require(end.debt() == 0, "D3MHub/end-debt-already-set");
+            EndLike _end = end;
+            require(_end.debt() == 0, "D3MHub/end-debt-already-set");
             require(ilks[ilk].culled == 0, "D3MHub/module-has-to-be-unculled-first");
-            _unwind(
+            _end.skim(ilk, address(_pool));
+            _fullUnwind(
                 ilk,
                 _pool,
-                type(uint256).max,
-                Mode.MCD_CAGED,
-                currentAssets
+                address(_end)
             );
-        } else if (ilks[ilk].tic != 0 || !ilks[ilk].plan.active()) {
-            _fix(ilk, address(_pool));
-
+        } else if (ilks[ilk].culled == 1) {
             // pool caged
-            _unwind(
+            _fullUnwind(
```

Comment:
The `pool caged` comment above should change to `pool culled` or dropped.

Reply:
Comment removed, it is pretty much self explanatory